File: solutions/javascript/darts/3/darts.js

export const score = (x, y) => {
  let radius = (( x ** 2) + ( y ** 2)) ** (0.5);
  return (radius <= 1) ? 10 
    : (radius > 1 && radius <= 5) ? 5  
    : (radius > 5 && radius <= 10) ? 1
    : 0;
}

File: solutions/ruby/darts/2/darts.rb
class Darts
  def initialize(x, y)
    self.radius = Math.sqrt(x ** 2 + y ** 2)
  end

  attr_accessor :radius

  public

  def score
    case radius
    when 0..1 then 10
    when 1..5 then 5
    when 5..10 then 1
    else 0
    end
  end
end
